weed/server/filer_server_handlers_api.go

```diff
@@ -0,0 +1,40 @@
+package weed_server
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/chrislusf/seaweedfs/weed/filer"
+	"github.com/chrislusf/seaweedfs/weed/glog"
+)
+
+func (fs *FilerServer) apiHandler(w http.ResponseWriter, r *http.Request) {
+	request := r.FormValue("request")
+	apiRequest := filer.ApiRequest{}
+
+	err := json.Unmarshal([]byte(request), &apiRequest)
+	if err != nil {
+		glog.V(0).Infoln("failing to read request", r.RequestURI, request)
+		writeJsonError(w, r, http.StatusInternalServerError, err)
+	}
+	switch apiRequest.Command {
+	case "listDirectories":
+		res := filer.ListDirectoriesResult{}
+		res.Directories, err = fs.filer.ListDirectories(apiRequest.Directory)
+		if err != nil {
+			res.Error = err.Error()
+		}
+		writeJsonQuiet(w, r, http.StatusOK, res)
+	case "listFiles":
+		res := filer.ListFilesResult{}
+		limit := 1
```

Comment:
limit can always be 100. The apiRequest.FileName is more like "offset" in a file list query.